daemon/src/engine/framework/Rcon.h

```diff
@@ -0,0 +1,99 @@
+/*
+===========================================================================
+Daemon BSD Source Code
+Copyright (c) 2013-2016, Daemon Developers
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Daemon developers nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL DAEMON DEVELOPERS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+===========================================================================
+*/
+#ifndef COMMON_RCON_H_
+#define COMMON_RCON_H_
+
+#include <string>
+#include "common/Cvar.h"
+
+namespace Rcon {
+
+/*
+ * Secure RCon algorithm
+ */
+enum class Secure {
+	Unencrypted,        // Allow unencrypted rcon
+	EncryptedPlain,     // Require encryption
+	EncryptedChallenge, // Require encryption and challege check
+	Invalid             // Invalid protocol
+};
+
+/*
+ * RCon message sent from a client to a server
+ */
+class Message
```

Comment:
I'm not a fan of your approach here: it seems to me that Message does to much and there is no clear separation between what is received, errors, etc. From what I understand you did this to have all the Rcon code at the same place, right? Message should be a method-less structure and the various methods should be changed to functions operating on that structure.

For example Acceptable would become bool Rcon::IsAcceptable(const Message& message, std::string* error);
The code using that function might have to do a bit more work to create the answer from the error string, but it would still be cleaner.